src/ekat/ekat_assert.cpp

```diff
@@ -44,23 +47,33 @@ int get_default_fpes () {
 }
 
 void enable_fpes (const int mask) {
+#ifdef EKAT_ENABLE_FPE
```

Comment:
I'm wondering if we should throw an error if this fcn is called without `EKAT_ENABLE_FPE` being defined. The user might wrongfully assume its mask is doing something, while it's not.

Also, we should silence compiler warnings if EKAT_ENABLE_FPE is not defined, with the usual `(void) mask;`.

Reply:
An alternative to the error: should we make certain function declared only if `EKAT_ENABLED_FPE` is defined? The end result is the same, we just catch it at compile time.

Reply:
Maybe we can use `spdlog` to issue a run-time warning if these functions are called without `EKAT_ENABLE_FPE`. What do you think?

Agree about silencing compiler warnings (speaking of `spdlog`!)

Reply:
Leaving the functions out if `EKAT_ENABLE_FPE` is indeed a compile-time fix, but might cause CPP propagation up through caller code. If that's okay with you, I'll go ahead and move the guards outside those functions.

Reply:
It's true that apps might have code like
```
auto m = ekat::get_default_fpes();
ekat::enable_fpes(m);
```
and they might not want to wrap it in some `#ifdef`. So You're right, we should not conditionally compile these fcns, but we should instead just warn the user.

I would use printf, and not spdlog, since we would have to build a logger on the fly. It would have to be console only, and on all ranks. So it would basically be just like a printf.

Reply:
Any opinion on `stdout` vs `stderr`? Probably splitting hairs here. :-)

Reply:
I always print to stdout, but I guess stderr is more appropriate?